extensions/map-layers-auth/src/test/ArcGisAccessClient.test.ts

```diff
@@ -31,13 +44,55 @@ describe("ArcGisUtilities tests", () => {
     fakeAccessClient = undefined;
   });
 
-  const sampleOnPremiseMapServer = "https://dtlgeoarcgis.adtl.com/server/rest/services/NewYork/NewYork3857/MapServer";
-  const sampleOnPremiseMapServerRestUrl = "https://dtlgeoarcgis.adtl.com/portal/sharing/rest/";
+  const sampleOnPremiseHostName = "https://dtlgeoarcgis.adtl.com";
+  const sampleOnPremiseMapServer = `${sampleOnPremiseHostName}/server/rest/services/NewYork/NewYork3857/MapServer`;
+  const sampleOnPremiseMapServerRestUrl = `${sampleOnPremiseHostName}/portal/sharing/rest/`;
   const sampleOnPremiseMapServerAuthorizeUrl = `${sampleOnPremiseMapServerRestUrl}oauth2/authorize`;
-  const sampleOnPremiseMapServerFallbackAuthorizeUrl = "https://dtlgeoarcgis.adtl.com/server/sharing/rest/oauth2/authorize";
+  const sampleOnPremiseMapServerFallbackAuthorizeUrl = `${sampleOnPremiseHostName}/server/sharing/rest/oauth2/authorize`;
 
   const sampleOnlineAuthorize1 = "https://www.arcgis.com/sharing/rest/oauth2/authorize";
 
+  it("should find the matching enterprise client id ", async () => {
+
+    let clientId = (fakeAccessClient as any)!.getMatchingEnterpriseClientId(sampleOnPremiseHostName);
+    expect(clientId === fakeAccessClient?.arcGisEnterpriseClientIds![0].clientId);
```

Comment:
if there was no implementation, the test would still pass as it would return `undefined === undefined` and be `true`, you might want to add an expectation that `clientId` is not undefined.

Reply:
good point